Packet++/header/GvcpLayer.h

```diff
@@ -0,0 +1,772 @@
+		{
+			return netToHost16(getGvcpHeader()->dataSize);
+		}
+
+		/**
+		 * @brief Get the request ID from the header
+		 */
+		uint16_t getRequestId() const
+		{
+			return netToHost16(getGvcpHeader()->requestId);
+		}
+
+		/**
+		 * @brief Get the command from the header
+		 */
+		GvcpCommand getCommand() const
+		{
+			return static_cast<GvcpCommand>(netToHost16(getGvcpHeader()->command));
+		}
+
+		/**
+		 * @brief Verify the magic number in the header
+		 * @return true The magic number is valid
+		 */
+		bool verifyMagicNumber() const
+		{
+			return getGvcpHeader()->magicNumber == internal::kGvcpMagicNumber;
+		}
+
+		/**
+		 * @brief Check if the acknowledge is required from the header
+		 * @return true The acknowledge is required
+		 */
+		bool hasAcknowledgeFlag() const
+		{
+			constexpr GvcpFlag kAcknowledgeFlag = 0b0000001;
+			return (getGvcpHeader()->flag & kAcknowledgeFlag) == kAcknowledgeFlag;
+		}
+
+		// implement Layer's abstract methods
+		std::string toString() const override;
+
+		// implement Layer's abstract methods
+		size_t getHeaderLen() const override
+		{
+			return sizeof(GvcpRequestHeader);
+		}
+	};
+
+	/// @brief Gvcp acknowledge layer
+	class GvcpAcknowledgeLayer : public GvcpLayer
+	{
+	public:
+		/**
+		 * @brief Construct a new GvcpLayer object
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		GvcpAcknowledgeLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		/**
+		 * @brief Construct a new GvcpRequestLayer object
+		 * @param[in] data A pointer to the data including the header and the payload
+		 * @param[in] dataSize The size of the data in bytes
+		 */
+		GvcpAcknowledgeLayer(const uint8_t* data, size_t dataSize);
+
+		/**
+		 * @brief Construct a new GvcpAcknowledgeLayer object
+		 * @param[in] status The response status
+		 * @param[in] command The command
+		 * @param[in] payloadData A pointer to the payload data, optional
+		 * @param[in] payloadDataSize The size of the payload data in bytes, optional
+		 * @param[in] ackId The acknowledge ID, optional
+		 * @note all the parameters will be converted to the network byte order
+		 */
+		GvcpAcknowledgeLayer(GvcpResponseStatus status, GvcpCommand command, const uint8_t* payloadData = nullptr,
+		                     uint16_t payloadDataSize = 0, uint16_t ackId = 0);
+
+		/**
+		 * @brief Get the header object
+		 * @return GvcpAckHeader* A pointer to the header object
+		 */
+		GvcpAckHeader* getGvcpHeader() const
+		{
+			return reinterpret_cast<GvcpAckHeader*>(m_Data);  // the header is at the beginning of the data
+		}
+
+		/**
+		 * @return the response status from the header
+		 */
+		GvcpResponseStatus getStatus() const
+		{
+			return static_cast<GvcpResponseStatus>((netToHost16(getGvcpHeader()->status)));
```

Comment:
This method should handle cases where `getGvcpHeader()->status` is an unexpected value and return `GvcpResponseStatus::Unknown`